hopsworks-api/src/main/java/io/hops/hopsworks/api/featurestore/DataValidationResource.java

```diff
@@ -0,0 +1,136 @@
+import javax.ws.rs.GET;
+import javax.ws.rs.POST;
+import javax.ws.rs.Path;
+import javax.ws.rs.PathParam;
+import javax.ws.rs.Produces;
+import javax.ws.rs.core.Context;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+import javax.ws.rs.core.SecurityContext;
+import java.util.List;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+@Stateless
+@TransactionAttribute(TransactionAttributeType.NEVER)
+@JWTRequired(acceptedTokens={Audience.API}, allowedUserRoles={"HOPS_ADMIN", "HOPS_USER"})
+@AllowedProjectRoles({AllowedProjectRoles.DATA_OWNER, AllowedProjectRoles.DATA_SCIENTIST})
+public class DataValidationResource {
+  private static Logger LOGGER = Logger.getLogger(DataValidationResource.class.getName());
+  private static final String MAIN_CLASS = "io.hops.hopsworks.verification.Verification";
+  private static final String HOPS_VERIFICATION_JAR_TEMPLATE = "hdfs:///user" + org.apache.hadoop.fs.Path.SEPARATOR
+      + "%s" + org.apache.hadoop.fs.Path.SEPARATOR + "hops-verification-assembly-%s.jar";
+  
+  @EJB
+  private FeaturestoreFacade featurestoreFacade;
+  @EJB
+  private FeaturegroupController featuregroupController;
+  @EJB
+  private DataValidationController dataValidationController;
+  @EJB
+  private JWTHelper jwtHelper;
+  @EJB
+  private Settings settings;
+  
+  private Featurestore featurestore;
+  private String path2hopsverification;
+  
+  public DataValidationResource setFeatureStore(Integer featureStoreId) {
+    this.featurestore = featurestoreFacade.findById(featureStoreId);
+    return this;
+  }
+  
+  @PostConstruct
+  public void init() {
+    path2hopsverification = String.format(HOPS_VERIFICATION_JAR_TEMPLATE, settings.getSparkUser(),
+        settings.getHopsVerificationVersion());
+  }
+  
+  @POST
+  @Path("{featuregroupId}/rules")
+  @Produces(MediaType.APPLICATION_JSON)
+  public Response addValidationRules(ConstraintGroupDTO constraintGroups,
+      @PathParam("featuregroupId") Integer featureGroupId,
+      @Context SecurityContext sc) throws FeaturestoreException {
+    Users user = jwtHelper.getUserPrincipal(sc);
+    FeaturegroupDTO featureGroup = featuregroupController.getFeaturegroupWithIdAndFeaturestore(featurestore,
+        featureGroupId);
+  
+    String rulesPath = dataValidationController.writeRulesToFile(user, featurestore.getProject(), featureGroup,
+        constraintGroups.toConstraintGroups());
+    
+    DataValidationSettingsDTO settings = new DataValidationSettingsDTO();
+    settings.setValidationRulesPath(rulesPath);
+    settings.setExecutablePath(path2hopsverification);
+    settings.setExecutableMainClass(MAIN_CLASS);
+    LOGGER.log(Level.FINE, "Validation settings: " + settings);
+    return Response.ok().entity(settings).build();
+  }
+  
+  @GET
+  @Path("{featuregroupId}/rules")
+  @Produces(MediaType.APPLICATION_JSON)
+  public Response getValidationRules(@PathParam("featuregroupId") Integer featureGroupId,
```

Comment:
Can you add the `@ApiOperation` tag and a docstring for each endpoint so that Swagger picks it up? It shows a short string after each endpoint with the annotation: https://app.swaggerhub.com/apis-docs/logicalclocks/hopsworks-api/0.10.0

Reply:
Sure